### DISCUSSION TITLE
[bug] Cart Line price with automatic Order discount shows different price data if navigating back from checkout

### DISCUSSION THREAD

**Discussion by matt-meagher** [Shopify/storefront-api-feedback#224] [Help]
Hello!
I have an Order Discount configured that takes 10% off the cost if the basket total is more than $1 (This discount is for testing purposes). See below:
![image](https://github.com/Shopify/storefront-api-feedback/assets/112413786/3288ca9a-0ee2-41d6-a282-036b624798bf)

When I add a product to cart I see the correct price with the discount in `cost` of the cart. The `cost` of the cart line is unchanged:
![image](https://github.com/Shopify/storefront-api-feedback/assets/112413786/024ff588-608f-4287-a198-2cebe9ef980b)

If I navigate to checkout and then hit back in the browser. The cart line now shows the discounted cost in `cost.totalAmount`:
![image](https://github.com/Shopify/storefront-api-feedback/assets/112413786/547eee44-5731-4fe7-bc6c-0a77ae1ec7c0)

Any further cart action (updating the quantity, adding another product, etc) resets the above.
**Expected Behavior**: I would expect this is consistent. If this is an example where the cart data is updated with checkout data it still seems like Order Discount data should be consistent. 
Since this is order wide (and not product specific) I would guess the discounted price should only be reflected in the `cart.cost` and not per cart line object.

Currently using api version `2023-10`.

Thank you for any support!

**Comment by johncraigcole** [comment]
Hey @matt-meagher thanks for reporting. I was able to reproduce the issue and created an issue for it internally. Don't have a timeline on a fix, but I will follow up when I know more.
